src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -67,7 +69,7 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
```

Comment:
Would line 65 above throw? It would be executed by iOS and call BrokerSilentStrategy. https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/5da17a099ef18727f821aa13e3f8a426f0578a49/src/client/Microsoft.Identity.Client/Internal/Requests/Silent/BrokerSilentStrategy.cs#L42

Not sure but seems like before the refactoring, iOS would still execute the `BrokerSilentStrategy.ExecuteAsync` but maybe it would throw in a different place, like when it called `SendTokenRequestToBrokerAsync`.

Reply:
I dont think this called `BrokerSilentStrategy.ExecuteAsync` for ios before the refactoring, see code below.

![image](https://user-images.githubusercontent.com/30090357/134598131-e6827fc8-5d6c-4f07-8be2-d26a59d92dd1.png)

from https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2825/files#

Line 65 will throw but the catch block on line 72 will ignore it and simply return the error back to the dev. This is how it used to work

Reply:
_clientStrategy is "CacheSilentStrategy". So it will not go to the BrokerSilentStrategy